self-hosted-agent/helm/main.tf

```diff
@@ -0,0 +1,171 @@
+terraform {
```

Comment:
I think it would be nicer to split this file, it's doing a lot of stuff
maybe 
1. The IAM user stuff
2. the Helm Stuff
3. The data + EKS module (like inputs)

But any other split would also do